src/storage/mvcc/reader.rs

```diff
@@ -514,11 +512,34 @@ impl<'a> MvccReader<'a> {
         // A lot of MVCC versions of a single row to GC.
         props.max_row_versions > GC_MAX_ROW_VERSIONS_THRESHOLD
     }
+
+    fn get_mvcc_properties(&self, safe_point: u64) -> Option<MvccProperties> {
+        let collection = match self.snapshot.get_properties_cf(CF_WRITE) {
+            Ok(v) => v,
+            Err(_) => return None,
+        };
+        if collection.is_empty() {
+            return None;
+        }
+        // Aggregate MVCC properties.
+        let mut props = MvccProperties::new();
+        for (_, v) in &*collection {
+            let mvcc = match MvccProperties::decode(v.user_collected_properties()) {
+                Ok(v) => v,
+                Err(_) => return None,
+            };
+            // Filter out properties after safe_point.
+            if mvcc.min_ts > safe_point {
```

Comment:
We may reuse mvcc_properties to check how many rows in a region in the future, I think it is better to extract a filter for the aggregator.